core/common/src/main/java/alluxio/conf/Source.java

```diff
@@ -49,9 +50,14 @@
     SYSTEM_PROPERTY,
     /**
      * The property value is set by user during runtime (e.g., Configuration.set or through
-     * HadoopConf). This source has the highest priority.
+     * HadoopConf).
      */
     RUNTIME,
+    /**
+     * The property value is set by HDFS API side Alluxio URI (e.g. Zookeeper configuration
```

Comment:
Can we make this generic so that it could apply to any type of Alluxio URI, not just URIs from HDFS? I don't want to add an assumption that this will only ever be used through the HDFS API

Reply:
Do you suggest make Alluxio URI from native API also the highest priority?

That's what we want to do but hard to accomplish and may have a lot of issues if the configuration singleton is not solved.

but I also feel strange to say the Source == URI is the highest priority only in HDFS API.

If we change to Alluxio URI always bigger than configuration in HDFS API,
we do not need the Source == URI and I will remove it.

```
  /**
   * Sets connection configuration from URI.
   *
   * @param uri the Alluxio URI to get connection configuration from
   */
  private void setAlluxioConfigurationFromURI(AlluxioURI uri) {
    // For backward compatible
    Configuration.set(PropertyKey.ZOOKEEPER_ENABLED, isZookeeperMode());

    if (uri.isZookeeperURI()) {
      Configuration.set(PropertyKey.ZOOKEEPER_ENABLED, true);
      Configuration.set(PropertyKey.ZOOKEEPER_ADDRESS, uri.getAuthority());
    } else if (uri.getHost() != null) {
      Configuration.set(PropertyKey.MASTER_HOSTNAME, uri.getHost());
      Configuration.set(PropertyKey.MASTER_RPC_PORT, uri.getPort());
      Configuration.set(PropertyKey.ZOOKEEPER_ENABLED, false);
      if (Configuration.isSet(PropertyKey.ZOOKEEPER_ADDRESS)) {
        Configuration.unset(PropertyKey.ZOOKEEPER_ENABLED);
      }
    }
  }
```

Reply:
I'm only suggesting to update the javadoc comment. `Source` is an API that exists separate from `HDFS`, so it's good to define the types of sources in a general-purpose way even if they are currently used in only one specific way. When we support `Source.URI` for native API in the future, the `Source.URI` contract doesn't need to change.